### PR TITLE
feat: [CDS-77525]: listing page layout setup and the zero state entities for gitX webhooks

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Listing page layout setup for gitX web-hooks  with zero state entities.
- Routing setup for gitX webhooks events page

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/ba4f72c7-98da-4ae7-bd81-73c65a1cf835

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/pages/webhooks/Webhooks.tsx [hunk 1]
```diff
@@ -1,14 +1,140 @@
...
+            value: SortFields.AZ09
+          },
+          {
+            label: getString('ZA90'),
+            value: SortFields.ZA90
+          }
+        ]}
+        defaultSortOption={[SortFields.LastUpdatedAt, Sort.DESC]}
+        handleCustomSortChange={handleCustomSortChange}
+        filterType={'Environment'}
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.tsx:134]: Adding TODO for future reference
`filterType={'Environment'}`

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  font-size: 20px !important;
+  line-height: 28px !important;
+  color: var(--grey-800) !important;
+}
+
+.noWebhookText {
+  width: 400px;
+  padding: var(--spacing-large) 0 var(--spacing-xxlarge) !important;
+  font-size: var(--font-size-medium) !important;
+  line-height: 18px !important;
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.module.scss:31]: nit:   --font-size-medium: 18px;
    
line-height: var(--font-size-medium);

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  width: 550px !important;
+
+  > div[class*='bp3-dialog-header'] {
+    margin: 0px 0px 6px !important;
+    width: 330px;
+  }
+}
+
+.noWebhookHeading {
+  padding-top: 50px !important;
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.module.scss:20]: use css variables

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+.dialogStylesWebhook {
+  padding-top: 36px !important;
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.module.scss:9]: use css variables

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  position: relative;
+  width: 550px !important;
+
+  > div[class*='bp3-dialog-header'] {
+    margin: 0px 0px 6px !important;
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.module.scss:14]: same here

**Comment by udayvunnam** [comment]
There's already an empty state illustration "src/modules/10-common/images/EmptySearchResults.svg"

File: src/modules/75-cd/pages/webhooks/Webhooks.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.tsx:134]: added inline

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.module.scss:31]: Done

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.module.scss:20]: done

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss (hunk 4 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.module.scss]: done

File: src/modules/75-cd/pages/webhooks/Webhooks.module.scss (hunk 5 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.module.scss:14]: done

**Comment by vikrant-harness** [comment]
Coverage for the new files will be added as a separate PR altogether

**Review by PratyushGar — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Comment by vikrant-harness** [comment]
The cypress checks are flaky and have been reported for fixing , the code is behind FF and is needed in develop for upcoming PR's

**Pull request merged**